keycloak/client.go

```diff
@@ -80,22 +112,36 @@ func (c Client) PutGroup(ctx context.Context, group Group) (Group, error) {
 		}
 	}
 
+	membErr := MembershipSyncErrors{}
+
 	for _, fm := range foundMemb {
 		if !contains(group.Members, *fm.Username) {
 			// user is not in group remove it
 			err := c.Client.DeleteUserFromGroup(ctx, token.AccessToken, c.Realm, *fm.ID, *found.ID)
 			if err != nil {
-				return res, err
+				membErr = append(membErr, MembershipSyncError{
+					Err:      err,
+					Username: *fm.Username,
+					Event:    UserRemoveError,
+				})
+				continue
 			}
 		} else {
 			res.Members = append(res.Members, *fm.Username)
 		}
 	}
 	newMemb := diff(group.Members, res.Members)
 
-	addedMemb, err := c.addUsersToGroup(ctx, token, *found.ID, newMemb)
+	addedMemb, addMembErr := c.addUsersToGroup(ctx, token, *found.ID, newMemb)
 	res.Members = append(res.Members, addedMemb...)
-	return res, err
+	if addMembErr != nil {
+		membErr = append(membErr, *addMembErr...)
+	}
```

Comment:
Alternative idea: Add the append function to the `MembershipSyncErrors` type, e.g. (prosa)
```go
func (mse *MembershipSyncErrors) AppendIfNotNil(err MembershipSyncError) {
  if err != nil {
    *mse = append(*mse, err)
  }
}
```

Reply:
alright :)